### PR TITLE
[42185] Issues when manually changing dates in datepicker

### PR DESCRIPTION
**Pull request by HDinger**
**Issue 1**
Manually inputting a start date that is later than the finish date moves the 'finish date' selection on the mini calendar instead. (The date picker allows save, and only shows an error afterwards.

_Expected behaviour: The finish date takes the new (later) start date_

**Issue 2**
Inputting a date in the future (start or finish) does not move the mini calendar to those days. The dates are input correctly but the user has no overview until they re-open the date picker after saving selection. 

_Expected behaviour: When manually inputing a date, the mini calendar moves to the month that is input (even if that is in the past or the future)._

Fixes https://community.openproject.org/projects/openproject/work_packages/42185#activity-20

### CODE REVIEW

**Comment by psatyal** [comment]
Hi @HDinger. 

Thank you. 

So issue #2 is fixed; the date picker is reacting instantly to typed dates.
As for issue #1, the delay greatly helps. In the example below:

- Entering later start dates later than the finish date consistently pushes the finish date also to this later date. (This is correct behaviour).
- The delay makes the interaction much better, and entering finish dates dates like "16" possible without having the start date reset to 1 (as seen in the gif below), but I think we can even increase the delay further. What value are we at now? What if we double it?

Screencapture:

dates-delay

Cheers
Parimal

**Review by psatyal — CHANGES_REQUESTED** [review]
Hi @HDinger. Can we try more like 600 or 800 ms delay?

File: frontend/src/app/shared/components/datepicker/datepicker.modal.service.ts [hunk 1]
```diff
@@ -209,15 +209,29 @@ export class DatepickerModalService {
 
     /* eslint-disable no-param-reassign */
```
Comment (b12f) [inline frontend/src/app/shared/components/datepicker/datepicker.modal.service.ts]: We can now reduce the scope of this disable comment to only the `if` on line 218

File: frontend/src/app/shared/components/datepicker/datepicker.modal.ts [hunk 2]
```diff
@@ -142,6 +148,15 @@ export class DatePickerModalComponent extends OpModalComponent implements AfterV
         this.initializeDatepicker(this.minimalDateFromPrecedingRelationship(relation));
         this.onDataChange();
       });
+
+    this
+      .dateChangedManually$
+      .pipe(
+        debounceTime(800),
```
Comment (b12f) [inline frontend/src/app/shared/components/datepicker/datepicker.modal.ts:161]: I'd put a comment either here or in the commit explaining why this is 800ms

**Review by b12f — APPROVED** [review]
LGTM code-wise, though I'd appreciate the explanatory comment around the magic debounce number.

**Comment by HDinger** [comment]
Merging as the failing specs are known flickerer

**Pull request merged**